### ISSUE TITLE
Force GPU memory limit

### ISSUE THREAD

**Issue by QingtaoLi1** [dotnet/TorchSharp#544]
As mentioned in [this question](https://stackoverflow.com/questions/49529372/force-gpu-memory-limit-in-pytorch), PyTorch now supports limiting GPU memory usage, which helps better management and planning on it.

**Comment by NiklasGustafsson** [comment]
I'm searching for the C++ code that implements it. Haven't found it yet.

**Comment by dayo05** [comment]
@NiklasGustafsson I found method implementation of set_per_process_memory_fraction is exists on torch/csrc/cuda/Module.cpp::_cuda_setMemoryFraction
I don't know about how does pytorch implementing native method to python, but this might be help :D

**Comment by dayo05** [comment]
Maybe this method is doing that: c10::cuda::CUDACachingAllocator::setMemoryFraction(fraction, device);

**Comment by dayo05** [comment]
> > CUDACachingAllocator
> 
> Which header file is that declared in? I don't find it in torch/cuda.h

[c10/cuda/CUDACachingAllocator.cpp](https://github.com/pytorch/pytorch/blob/224ae0da107ee426a2e19dab3eee52b6252f842f/c10/cuda/CUDACachingAllocator.cpp) here in 927 line

**Comment by dayo05** [comment]
@NiklasGustafsson I'm trying to implement this method but, this method is only available on CUDA-backend api. Is there any place to locate cuda-only-apis?

**Comment by NiklasGustafsson** [comment]
No, because the interop layer is backend-independent, so it doesn't link to anything that isn't available in both.
